kapitan/initialiser.py

```diff
@@ -20,27 +20,150 @@
+                path = self.get_new_target_path("%s.yml" % i)
+                self.dump_yaml(target_file,path)
+        else:
+            path = self.get_new_target_path(self.target_template_name)
+            self.dump_yaml(target_file, path)
+
+    def copy_component_file(self):
+        """
+            copy component file `my_component.yml` to new directory
+            only copy compile_obj's input_type is in compile-input
+            if compile-input is not provided my user copy all compile_obj
+            append input_paths of accepted compile_obj to copy_path_list
+        """
+
+        component_file = self.get_component_template()
+        compile_objs = []
+        for compile_obj in component_file['parameters']['kapitan']['compile']:
+            if compile_obj['input_type'] in self.compile_inputs or len(self.compile_inputs) == 0:
+                self.copy_path_list += compile_obj['input_paths']
+                compile_objs.append(compile_obj)
+        component_file['parameters']['kapitan']['compile'] = compile_objs
+
+        path = self.get_new_component_path()
+        self.dump_yaml(component_file, path)
+
+    def copy_compile_components(self):
+        """
+            copy for files present in copy_path_list to new location
+        """
+        for path in self.copy_path_list:
+            self.copy_components(path)
+
+    def get_new_target_path(self,name):
+        path = os.path.join(self.directory, self.target_dir)
+        self.handle_directory_creation(path)
+        path = os.path.join(path,name)
+        return path
+
+    def get_new_component_path(self):
+        path = os.path.join(self.directory, self.component_dir)
+        self.handle_directory_creation(path)
+        path = os.path.join(path,self.component_template_name)
+        return path
+
+    def handle_directory_creation(self,path):
+        try:
+            os.makedirs(path, exist_ok=True)
+        except OSError as e:
+            pass
+
+    def copy_components(self,path):
+        """
+            copy allowed components to new location
+            if directory then copy every entire directory
+        """
+        template_file_path = os.path.join(self.templates_directory, path)
+        new_file_path = os.path.join(self.directory, path)
+        if os.path.isdir(template_file_path):
+            new_file_path = new_file_path.rstrip('/')
+            dirname = os.path.dirname(new_file_path)
+            self.handle_directory_creation(dirname)
+            shutil.copytree(template_file_path, new_file_path)
+        else:
+            file_directory = os.path.dirname(new_file_path)
+            self.handle_directory_creation(file_directory)
+            shutil.copy(template_file_path, new_file_path)
+
+
+
+    def dump_yaml(self,obj,path):
+        with open(path, 'w') as outfile:
+            yaml.dump(obj, outfile, default_flow_style=False)
+
+    def list_new_directory(self):
+        """
+            list all files in new location
```

Comment:
in self.directory* And rename list_new_directory to just `list_directory` if that's the actual function.